dbms/src/Functions/tests/gtest_cast_json_as_string.cpp

```diff
@@ -27,6 +29,49 @@ namespace DB::tests
 {
 class TestCastJsonAsString : public DB::tests::FunctionTest
 {
+public:
+    ColumnWithTypeAndName executeCastJsonAsStringFunction(
+        Context & context,
+        const ColumnWithTypeAndName & input_column,
+        const tipb::FieldType & field_type)
+    {
+        auto & factory = FunctionFactory::instance();
+        ColumnsWithTypeAndName columns({input_column});
+        ColumnNumbers argument_column_numbers;
+        for (size_t i = 0; i < columns.size(); ++i)
+            argument_column_numbers.push_back(i);
+
+        ColumnsWithTypeAndName arguments;
+        for (const auto argument_column_number : argument_column_numbers)
+            arguments.push_back(columns.at(argument_column_number));
+
+        const String func_name = "cast_json_as_string";
+        auto builder = factory.tryGet(func_name, context);
+        if (!builder)
+            throw TiFlashTestException(fmt::format("Function {} not found!", func_name));
+        auto func = builder->build(arguments, nullptr);
+        auto * function_build_ptr = builder.get();
+        if (auto * default_function_builder = dynamic_cast<DefaultFunctionBuilder *>(function_build_ptr);
+            default_function_builder)
+        {
+            auto * function_impl = default_function_builder->getFunctionImpl().get();
+            if (auto * function_cast_json_as_string = dynamic_cast<FunctionsCastJsonAsString *>(function_impl);
+                function_cast_json_as_string)
+            {
+                function_cast_json_as_string->setOutputTiDBFieldType(field_type);
+            }
+            else
+            {
+                throw TiFlashTestException(fmt::format("Function {} not found!", func_name));
+            }
+        }
```

Comment:
Seems useless because `DAGExpressionAnalyerHelper` will be called when `raw_function_test` is false

Reply:
Can't get your point here, just introduce this method for test to set tidb field type here